### PR TITLE
Add storage name option to entity attributes

### PR DESCRIPTION
**Pull request by letstalk-nl**
### 1. Why is this change necessary?
As discussed in Slack, there is a need for some consistency with the previous way of creating definitions where you can influence the storagename.

### 2. What does this change do, exactly?
Add the option to change the storage name for fields defined with entity attributes.

### 4. Please link to the relevant issues (if any).
https://shopwarecommunity.slack.com/archives/C011VFQT7GB/p1718088460270309

### 5. Checklist

- [ ] I have rebased my changes to remove merge conflicts
- [x] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by letstalk-nl** [comment]
I think the docs should be updated here; https://github.com/shopware/docs/blob/d9cd7b8e82c256cab8c6eca03e27065b8da3b8f9/guides/plugins/plugins/framework/data-handling/entities-via-attributes.md

Not sure how this should be done in combination with this PR.

**Comment by letstalk-nl** [comment]
> I think the docs should be updated here; https://github.com/shopware/docs/blob/d9cd7b8e82c256cab8c6eca03e27065b8da3b8f9/guides/plugins/plugins/framework/data-handling/entities-via-attributes.md
> 
> Not sure how this should be done in combination with this PR.

Also might be relevant to add documentation on the reference field attribute, as this seems now necessary to have automatic updates during updates?

File: src/Core/Framework/DataAbstractionLayer/Attribute/Field.php [hunk 1]
```diff
@@ -17,7 +17,7 @@ public function __construct(
         public string $type,
         public bool $translated = false,
         public bool|array $api = false,
-        public ?string $column = null,
+        public ?string $storageName = null,
```
Comment (mitelg) [inline src/Core/Framework/DataAbstractionLayer/Attribute/Field.php:20]: this is a breaking change, as this is a public property. I am not sure about status of the whole attribute entity feature (if it is still declared as experimental or so), but it was already released and therefore, we need to consider this

Comment (letstalk-nl) [reply src/Core/Framework/DataAbstractionLayer/Attribute/Field.php:20]: I'm hoping it is still marked as experimental as we've had enough issues with it to warrant several pull requests 😅

Also unsure of the state of this PR as I've had major merge conflicts with trunk that seemed to resolve the one of the issues I'm trying to solve in this PR. Maybe best to wait until Oliver is back from holiday to resolve all this.

**Comment by OliverSkroblin** [comment]
Hi, 
sorry I was in vacation. PR was merged: https://github.com/shopware/shopware/commit/c0db1a5ab3f398ae7d4ddc61b5f6f7d43d54cfca

**Comment by OliverSkroblin** [comment]
Sorry my mistake. Thought it was the MR for the storage name declaration. Reopened

**Comment by OliverSkroblin** [comment]
So storage name is already support: https://github.com/shopware/shopware/commit/c0db1a5ab3f398ae7d4ddc61b5f6f7d43d54cfca

What is missing? State machine field here?

Would be this [commit](https://github.com/shopware/shopware/pull/3766/commits/78f1168eac0c73d1e670da6e13676edf355d5a25) right?

**Comment by letstalk-nl** [comment]
@OliverSkroblin indeed that commit, and also the ability to configure property name on the version field, [this commit](https://github.com/shopware/shopware/pull/3766/commits/35a803de56a045e5695c8425ca064d000d61d6e9#diff-fc8659176b4e38fdd5b9cda01a8fbc5cfe0f14f2e6262c8da18ddc233aa4baf2)

**Comment by OliverSkroblin** [comment]
Thank you for all your improvements! All changes are now in the current trunk. The state machine field was added with [this commit](https://github.com/shopware/shopware/commit/2fcdc05740242cac05091935b3ec854f60a80ee7)

**Pull request closed without merging**